### PR TITLE
add KeyCharacterMap.UnavailableException

### PR DESCRIPTION
**Pull request by skyseraph**
add KeyCharacterMap.UnavailableException in charMap

### CODE REVIEW

File: server/src/main/java/com/genymobile/scrcpy/EventController.java [hunk 1]
```diff
@@ -19,7 +19,8 @@
     private final Device device;
     private final DesktopConnection connection;
 
-    private final KeyCharacterMap charMap = KeyCharacterMap.load(KeyCharacterMap.VIRTUAL_KEYBOARD);
+    // private final KeyCharacterMap charMap = KeyCharacterMap.load(KeyCharacterMap.VIRTUAL_KEYBOARD);
```
Comment (npes87184) [inline server/src/main/java/com/genymobile/scrcpy/EventController.java]: Not required line.

**Comment by rom1v** [comment]
Hi,

Thank you for your contribution.

However, could you explain what problem it solves, please? Do you have a device where there is no VIRTUAL_KEYBOARD `KeyCharacterMap`?

If that's the case, it will still fails on every call to `injectChars()`, won't it?

**Comment by skyseraph** [comment]
problem in Android Q  pixel

**Comment by rom1v** [comment]
> problem in Android Q pixel

But then, every `injectChars()` will fail, right?

**Comment by skyseraph** [comment]
> > problem in Android Q pixel
> 
> But then, every `injectChars()` will fail, right?

**Comment by skyseraph** [comment]
> > problem in Android Q pixel
> 
> But then, every `injectChars()` will fail, right?

all exception  ~

**Comment by rom1v** [comment]
In seems everything works fine on Android Q for injecting characters.

**Comment by yume-chan** [comment]
Saw this on an Android Q device:

(Scrcpy 3.3.4, don't have framework.jar)

```
07-25 15:19:27.576.000001  2560  2560 W app_process: type=1400 audit(0.0:963): avc: denied { read } for name="u:object_r:mm_video_prop:s0" dev="tmpfs" ino=19614 scontext=u:r:shell:s0 tcontext=u:object_r:mm_video_prop:s0 tclass=file permissive=0V
07-25 15:19:27.593.17515  2560  2560 E libc    : Access denied finding property "ro.vendor.net.upload.benchmark.default"
07-25 15:19:27.593.252338  2560  2560 E libc    : Access denied finding property "persist.vendor.debug.en.drpcrpt"
07-25 15:19:27.576.000001  2560  2560 W app_process: type=1400 audit(0.0:964): avc: denied { read } for name="u:object_r:vendor_default_prop:s0" dev="tmpfs" ino=19672 scontext=u:r:shell:s0 tcontext=u:object_r:vendor_default_prop:s0 tclass=file permissive=0H
07-25 15:19:27.595.880567  2560  2560 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 2000 <<<<<<
07-25 15:19:27.595.880567  2560  2560 D AndroidRuntime: 
07-25 15:19:27.586.000001  2560  2560 W app_process: type=1400 audit(0.0:965): avc: denied { read } for name="u:object_r:device_config_runtime_native_boot_prop:s0" dev="tmpfs" ino=19558 scontext=u:r:shell:s0 tcontext=u:object_r:device_config_runtime_native_boot_prop:s0 tclass=file permissive=0
07-25 15:19:27.600.298744  2560  2560 E libc    : Access denied finding property "persist.device_config.runtime_native_boot.enable_apex_image"
07-25 15:19:27.600.443015  2560  2560 I AndroidRuntime: Using default boot image
07-25 15:19:27.600.587338  2560  2560 E libc    : Access denied finding property "persist.device_config.runtime_native_boot.disable_lock_profiling"
07-25 15:19:27.600.615567  2560  2560 I AndroidRuntime: Leaving lock profiling enabled
07-25 15:19:27.586.000001  2560  2560 W app_process: type=1400 audit(0.0:966): avc: denied { read } for name="u:object_r:device_config_runtime_native_boot_prop:s0" dev="tmpfs" ino=19558 scontext=u:r:shell:s0 tcontext=u:object_r:device_config_runtime_native_boot_prop:s0 tclass=file permissive=0
07-25 15:19:27.586.000001  2560  2560 W app_process: type=1400 audit(0.0:967): avc: denied { read } for name="u:object_r:device_config_runtime_native_boot_prop:s0" dev="tmpfs" ino=19558 scontext=u:r:shell:s0 tcontext=u:object_r:device_config_runtime_native_boot_prop:s0 tclass=file permissive=0
07-25 15:19:27.600.934317  2560  2560 E libc    : Access denied finding property "persist.device_config.runtime_native_boot.enable_generational_cc"
07-25 15:19:27.706.254056  2560  2560 D ICU     : Time zone APEX file found: /apex/com.android.tzdata/etc/icu/icu_tzdata.dat
07-25 15:19:27.733.000983  2560  2560 I VivoSystemFrameworkFactory: Instance implClass com.vivo.VivoSystemFrameworkFactoryImpl
07-25 15:19:27.751.093119  2560  2560 W app_process: JNI RegisterNativeMethods: attempt to register 0 native methods for android.media.AudioAttributes
07-25 15:19:27.760.259212  2560  2560 D AndroidRuntime: Calling main entry com.genymobile.scrcpy.Server
07-25 15:19:27.769.943067  2560  2560 I scrcpy  : Device: [vivo] vivo V2034A (Android 10)
07-25 15:19:27.805.85989  2560  2560 D VivoConfigStore: hasVivoFeature key=vivo.software.globaltheme from property
07-25 15:19:27.806.766869  2560  2560 I FtBuild : getRomVersion mRomVer=11.5
07-25 15:19:27.808.320671  2560  2560 D VivoConfigStore: hasVivoFeature key=vivo.software.pem.ddc from property
07-25 15:19:27.816.806973  2560  2560 W asset   : unable to execute idmap2: Permission denied
07-25 15:19:27.816.845306  2560  2560 W AssetManager: 'idmap2 --scan' failed: no static="true" overlays targeting "android" will be loaded
07-25 15:19:27.806.000001  2560  2560 W main    : type=1400 audit(0.0:968): avc: denied { execute } for name="idmap2" dev="dm-2" ino=2154 scontext=u:r:shell:s0 tcontext=u:object_r:idmap_exec:s0 tclass=file permissive=0
07-25 15:19:27.806.000001  2560  2560 W main    : type=1400 audit(0.0:969): avc: denied { search } for name="bbkcore" dev="sda20" ino=1712 scontext=u:r:shell:s0 tcontext=u:object_r:vivo_data_file:s0 tclass=dir permissive=0�
07-25 15:19:27.806.000001  2560  2560 W main    : type=1400 audit(0.0:970): avc: denied { search } for name="bbkcore" dev="sda20" ino=1712 scontext=u:r:shell:s0 tcontext=u:object_r:vivo_data_file:s0 tclass=dir permissive=0�
07-25 15:19:27.826.747129  2560  2560 I SystemServiceRegistry: regMtkService start
07-25 15:19:27.826.937546  2560  2560 W System  : ClassLoader referenced unknown path: system/framework/mediatek-framework.jar
07-25 15:19:27.827.239369  2560  2560 E SystemServiceRegistry: regMtkService:java.lang.ClassNotFoundException: mediatek.app.MtkSystemServiceRegistry
07-25 15:19:27.827.287181  2560  2560 I SystemServiceRegistry: setMtkSystemServiceName start
07-25 15:19:27.827.306244  2560  2560 I SystemServiceRegistry: registerAllMtkService start
07-25 15:19:27.827.417546  2560  2560 D VivoSystemServiceRegistry: registerVivoSystemService
07-25 15:19:27.828.564785  2560  2560 D VivoConfigStore: hasVivoFeature key=vivo.software.doubleinstancestoragepath from property
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : Could not load key character map for device -1
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : android.view.KeyCharacterMap$UnavailableException: Could not load key character map for device -1
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : 	at android.view.KeyCharacterMap.load(KeyCharacterMap.java:339)
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : 	at com.genymobile.scrcpy.control.Controller.<init>(Controller.java:86)
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : 	at com.genymobile.scrcpy.Server.scrcpy(Server.java:114)
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : 	at com.genymobile.scrcpy.Server.internalMain(Server.java:271)
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : 	at com.genymobile.scrcpy.Server.main(Server.java:215)
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:372)
07-25 15:19:27.872.554994  2560  2560 E scrcpy  : 
07-25 15:19:27.874.480879  2560  2560 I app_process: System.exit called, status: 1
07-25 15:19:27.874.54265  2560  2560 I AndroidRuntime: VM exiting with result code 1.
07-25 15:19:27.939.692494  2604  2604 E libc    : Access denied finding property "ro.vendor.net.upload.benchmark.default"
07-25 15:19:27.926.000001  2604  2604 W logcat  : type=1400 audit(0.0:972): avc: denied { read } for name="u:object_r:vendor_default_prop:s0" dev="tmpfs" ino=19672 scontext=u:r:shell:s0 tcontext=u:object_r:vendor_default_prop:s0 tclass=file permissive=0.
```

**Pull request closed without merging**